Makefile

```diff
@@ -1,6 +1,6 @@
 SHELL=/bin/bash
 
-ifneq ($(shell python -c "import sys; print(hasattr(sys, 'real_prefix'))"),True)
+ifneq ($(shell python -c "import sys; print(hasattr(sys, 'real_prefix') or sys.base_prefix != sys.prefix)"),True)
```

Comment:
Azul uses a separate target, while this test runs every time the Makefile is evaluated. There are pros and cons to each. I considered changing this to match Azul, but there is a lot of overhead involved. All other targets would then need to depend on the `check_venv` target. 

I wanted to keep my changes minimally invasive, but the exact method used here doesn't work on later versions of `virtualenv`. That's why I just updated this check.

Reply:
I meant the actual check, not how it's invoked. This bit:

https://github.com/DataBiosphere/azul/blob/26a6a6d5e2a0c9d73ba941b039a36fa6b9e80c08/common.mk#L16

Reply:
I think checking `$VIRTUAL_ENV` would be the best method just because it's the most clear and obvious. It has one caveat that the other method isn't susceptible to:

> The problem is, this only works when the environment is activated by the activate shell script.
>
>You can start the environment's scripts without activating the environment, so if that is a concern, you have to use a different method.

but I don't think that's a realistic concern.

https://stackoverflow.com/a/58026969/7830612 has a good summary of the differences.

Let me know if you want me to switch to checking `$VIRTUAL_ENV`.

Reply:
For the record, we decided to use VIRTUAL_ENV during one of the stand-ups. Anything preventing you from implementing that option?